smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/HttpProtocolClientGenerator.kt

```diff
@@ -49,19 +48,23 @@ open class HttpProtocolClientGenerator(
                 ServiceGenerator.renderOperationDefinition(model, serviceShape, symbolProvider, writer, operationsIndex, it)
                 writer.openBlock(" {", "}") {
                     val operationStackName = "operation"
-                    val generator = MiddlewareExecutionGenerator(ctx, writer, httpBindingResolver, httpProtocolCustomizable, operationMiddleware, operationStackName)
-                    generator.render(serviceShape, it) { writer, labelMemberName ->
-                        writer.write(
-                            "throw \$N.unknownError(\"uri component \$L unexpectedly nil\")",
-                            SmithyTypes.ClientError,
-                            labelMemberName,
-                        )
-                    }
-
+                    val generator = makeMiddlewareExecutionGenerator(ctx, writer, httpBindingResolver, httpProtocolCustomizable, operationMiddleware, operationStackName)
+                    generator.render(serviceShape, it)
                     writer.write("return try await op.execute(input: input)")
                 }
                 writer.write("")
             }
         }
     }
+
+    open fun makeMiddlewareExecutionGenerator(
```

Comment:
This is overridden in a AWS-specific subclass to allow for AWS-specific customization.